schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/dao/common/ActivityPackageActivityDao.java

```diff
@@ -0,0 +1,49 @@
+package com.jasify.schedule.appengine.dao.common;
+
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.Query;
+import com.jasify.schedule.appengine.dao.BaseCachingDao;
+import com.jasify.schedule.appengine.dao.BaseDaoQuery;
+import com.jasify.schedule.appengine.meta.activity.ActivityPackageActivityMeta;
+import com.jasify.schedule.appengine.model.EntityNotFoundException;
+import com.jasify.schedule.appengine.model.activity.Activity;
+import com.jasify.schedule.appengine.model.activity.ActivityPackageActivity;
+import com.jasify.schedule.appengine.model.activity.ActivityType;
+import org.slim3.datastore.CompositeCriterion;
+import org.slim3.datastore.Datastore;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * @author szarmawa
+ * @since 09/06/15.
+ */
+public class ActivityPackageActivityDao extends BaseCachingDao<ActivityPackageActivity> {
+    private final ActivityTypeDao activityTypeDao = new ActivityTypeDao();
+
+    public ActivityPackageActivityDao() {
+        super(ActivityPackageActivityMeta.get());
+    }
+
+    public List<ActivityPackageActivity> getBy(final Activity activity) {
+        ActivityPackageActivityMeta meta = getMeta();
+        try {
+            ActivityType activityType = activityTypeDao.get(activity.getActivityTypeRef().getKey());
+            final Key organisationKey = activityType.getOrganizationRef().getKey();
+            return query(new BaseDaoQuery<ActivityPackageActivity, ActivityPackageActivityMeta>(meta, new Serializable[0]) {
+                @Override
+                public List<Key> execute() {
+                    return Datastore.query(meta, organisationKey)
+                            .filter(new CompositeCriterion(meta,
+                                    Query.CompositeFilterOperator.AND,
+                                    meta.activityRef.equal(activity.getId()))).asKeyList();
+                }
```

Comment:
1. Ancestor query should be transactional, if you call this method with an active transaction, the `activityTypeDao.get` will be in the transaction, but the query won't.
 2. I don't understand why you need composite criterion here, what are you "and"ing?

This could be written as:
```java
Datastore
   .query(Datastore.getCurrentTransaction(), meta, organizationKey)
   .filter(meta.activityRef.equal(activity.getId())
   .asKeyList();
```